src/api/app/models/badge.rb

```diff
@@ -0,0 +1,134 @@
+class Badge
```

Comment:
I'm still not convinced that we need to maintain all this code for a relatively fixed amount of badges. There are only 12 possible build results for a package. There are only 2 possible build result states per project. Sounds like completely static badges for packages to me. And for projects it sounds like a badge template with some text replacement.